plugins/ingest/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -337,12 +353,28 @@ private void setFieldValue(String path, Object value, boolean append) {
         }
     }
 
-    public String getMetadata(MetaData metaData) {
-        return this.metaData.get(metaData.getFieldName());
+    public String getEsMetadata(MetaData esMetadata) {
+        return this.esMetadata.get(esMetadata.getFieldName());
     }
 
-    public void setMetaData(MetaData metaData, String value) {
-        this.metaData.put(metaData.getFieldName(), value);
+    public Map<String, String> getEsMetadata() {
```

Comment:
I think we should have two maps. One map holding es metadata and source, and the other map holding the ingest metadata. This is going to fit nicely with the way we're going to refer to source and metadata fields. (just `_id` for id of the doc, `field1` to get a field name field1 from the source. Since meta fields can't be used in the source this should work out) The `IngestDocument.MetaData` enum can used to detect if a top level entry is either part of the source or metadata.